### PR TITLE
Allow Any Postgres Params

### PR DESCRIPTION
**Pull request by eberlep**

### CODE REVIEW

**Comment by eberlep** [comment]
Overwriting of default params works as well:

Test (Control Plane):
```yaml
 spec:
   postgresParams:
     pgaudit.log: all, -misc
     pgaudit.log_parameter: "off"
```


Result (Service Cluster):
```yaml
 spec:
   postgresql:
     parameters:
       pgaudit.log: all, -misc
       pgaudit.log_catalog: "off"
       pgaudit.log_parameter: "off"
       pgaudit.log_relation: "on"
       shared_buffers: 64MB
       shared_preload_libraries: pgaudit
```
Default (Service Cluster):
```yaml
 spec:
   postgresql:
     parameters:
       pgaudit.log: ddl
       pgaudit.log_catalog: "off"
       pgaudit.log_parameter: "on"
       pgaudit.log_relation: "on"
       shared_buffers: 64MB
       shared_preload_libraries: pgaudit
```

**Comment by eberlep** [comment]
Setting numeric values as strings seems to work as well:

Test (Control Plane):
```yaml
 spec:
   postgresParams:
     max_connections: "1001"
```

Result (Service Cluster):
```yaml
 spec:
   postgresql:
     parameters:
       max_connections: "1001"
```

Result (Database Instance):
```
postgres=# SHOW max_connections;
 max_connections 
-----------------
 1001
(1 row)
```

File: api/v1/postgres_types.go [hunk 1]
```diff
@@ -653,3 +664,10 @@ func enableAuditLogs(parameters map[string]string) {
 	parameters["pgaudit.log_relation"] = "on"
 	parameters["pgaudit.log_parameter"] = "on"
 }
+
+// setPostgresParams add the provided params to the parameter map
+func setPostgresParams(parameters map[string]string, providedParams map[string]string) {
+	for k, v := range providedParams {
```
Comment (majst01) [inline api/v1/postgres_types.go:670]: My feeling is that there should be at least be a denylist of params be configurable which a user cannot override

**Review by majst01 — COMMENTED** [review]
parameters which are not allowed to change should be considered, at least in a slice in the configmap

File: api/v1/postgres_types.go (hunk 1 above)
Comment (eberlep) [reply api/v1/postgres_types.go:670]: In the central `cloud-api` or locally in each postgreslet/service cluster? Locally is more flexible, but also more difficult to maintain.

Comment (majst01) [reply api/v1/postgres_types.go:670]: unsure about this

Comment (majst01) [reply api/v1/postgres_types.go:670]: implementation wise, locally in a configmap is easier, deployment is done via playbook, so keeping this configmap identical between installation should also be a easy task

Comment (eberlep) [reply api/v1/postgres_types.go:670]: I used viper to read a (apparently space-separated) list of blocked values from the environment, convert it into a map for easier use and pass that map along to the code in question. now before setting a given postgres param, we check if is contained in the blocklist. if it is, we simply ignore it.

**Review by majst01 — APPROVED** [review]

**Pull request merged**